src/main/java/gregtech/integration/jei/multiblock/MultiblockInfoPage.java

```diff
@@ -26,4 +38,51 @@ public float getDefaultZoom() {
                 .map(I18n::format)
                 .collect(Collectors.toList());
     }
+
+    /**
+     * Gets the Map containing all tooltips for MultiblockAbilities. Will generate the map if it does not exist
+     * @return - The Map containing Tooltips and Formatting for specific MultiblockAbilities
+     */
+    public Map<ItemStack, List<Tuple<String, TextFormatting>>> getAbilityTooltipMap() {
+        // Generate on first use
+        if (this.abilityTooltips.size() == 0) {
```

Comment:
isEmpty() is usually better - but since you are using an array list it won't make much difference here.

e.g. somebody could change it to use a LinkedList or something where size() is not a trivial operation

Reply:
Actually, this is an Object2ObjectOpenCustomHashMap which ironically does:
````
	public boolean isEmpty() {
		return size() == 0;
	}
````